storage/util/dynamic_percentile.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"fmt"
+	"math"
+	"sync"
+)
+
+// MetricType is the metric type DynamicPercentile algo expects.
+type MetricType int64
+
+// DynamicPercentile dynamically calculates the delay at a fixed percentile, based on
```

Comment:
Nit: wouldn't call it "delay" if the value is not typed as time.

Reply:
Not required, as changed the type.